### PR TITLE
Update Golang to 1.18 (from 1.18beta2)

### PR DESCRIPTION
**Pull request by yankay**
Since Golang 1.18 is released, Golang 1.18 was released: https://go.dev/blog/go1.18

Updating Golang to 1.18 (from 1.18beta2) makes sense.

Because [the constraints: move to x/exp for Go 1.18](https://github.com/golang/go/issues/50792), change to go import "constraints"

Refer: https://github.com/kubernetes-sigs/kpng/issues/274

Signed-off-by: Kay Yan <kay.yan@daocloud.io>

### CODE REVIEW

File: client/diffstore2/any-leaf.go [hunk 1]
```diff
@@ -1,8 +1,9 @@
 package diffstore2
 
 import (
-	"constraints"
 	"fmt"
+
+	"golang.org/x/exp/constraints"
```
Comment (mcluseau) [inline client/diffstore2/any-leaf.go:6]: Hi, I didn't check, but this change looks weird: constraints are out of stdlib now?

Comment (yankay) [reply client/diffstore2/any-leaf.go:6]: YES,  constraints are out of stdlib.

We can found the details at : https://github.com/golang/go/issues/50792, https://utcc.utoronto.ca/~cks/space/blog/programming/Go118NoConstraintsPackage

Comment (yankay) [reply client/diffstore2/any-leaf.go:6]: HI @mcluseau . Is the change good to merge ?

**Comment by yankay** [comment]
HI @mcluseau . Is the PR good to merge ?

**Comment by mcluseau** [comment]
sorry extremely hard to find time nowadays

/lgtm
/approve

**Pull request merged**